alacritty/src/config/monitor.rs

```diff
@@ -16,17 +16,20 @@ const DEBOUNCE_DELAY: Duration = Duration::from_millis(10);
 const DEBOUNCE_DELAY: Duration = Duration::from_millis(1000);
 
 pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventProxy) {
-    // Canonicalize all paths, filtering out the ones that do not exist.
-    paths = paths
-        .drain(..)
-        .filter_map(|path| match fs::canonicalize(&path) {
-            Ok(path) => Some(path),
-            Err(err) => {
-                error!("Unable to canonicalize config path {:?}: {}", path, err);
-                None
-            },
-        })
-        .collect();
+    // Also watch the targets of symbolic links
+    paths.append(
+        &mut paths
+            .iter()
+            .filter_map(|path| match fs::canonicalize(&path) {
+                Ok(canonical_path) if canonical_path != *path => Some(canonical_path),
```

Comment:
We maybe should make sure this is actually a symlink. This way it would also add a path twice if it contains segments like `/../`, which is unnecessary. So to ensure the number of watched paths is as small as possible, we should only add symlinks twice.

Reply:
Yes, and the current version is wrong anyway: we're not filtering the base paths anymore so we might end up watching a non-existent path.